src/resources/CaseAssist/CaseAssistInterfaces.ts

```diff
@@ -0,0 +1,44 @@
+import CaseAssist from './CaseAssist';
+
+export enum DocumentSuggestionsStrategies {
+    ITD = 'ITD',
+}
+
+export enum TypingAidsStrategies {
+    StaticValues = 'StaticValues',
+    ValuesFromIndex = 'ValuesFromIndex',
+}
+
+export interface CaseAssistListOptions {
+    page?: number;
+    pageSize?: number;
+    filter?: string;
+}
+
+interface DocumentSuggestionsConfigurations {
+    ITD: {
+        pipeline?: string;
+        fieldsToFeed: string[];
+        filter?: string;
+        strategy: DocumentSuggestionsStrategies;
+    };
+}
+
+interface TypingAidsConfigurations {
+    StaticValues: {
+        values: string[];
```

Comment:
Should we be adding a "type" in both configurations ? Something like `type: 'StaticValues` ? Otherwise, you'll get a return of `TypingAidsConfigurations['StaticValues'] | TypingAidsConfigurations['ValuesFromIndex'];` without a clean way to validate which one you are dealing with.

In your code, you'll be able to do something like:
```javascript
switch(typingAidsConfiguration.type) {
    case 'ValueFomrIndex':
        'do something;
        break;
    case 'StaticValues'
        'do something else'
        break;
}
```

and have auto completion working in TypeScript.

Reply:
Good idea

Reply:
Although i think the "strategy" key would work the same no?

```
switch(typingAidsConfiguration.strategy) {
    case 'ValueFomrIndex':
        'do something;
        break;
    case 'StaticValues'
        'do something else'
        break;
}
```

Reply:
Yup, I think you are right :)